### PR TITLE
New add-on - nfs-utils (nfsv4 support)

### PR DESCRIPTION
**Pull request by lrusak**
I wanted to try and incorporate this into OE core, but it would require using the util-linux mount command instead of the busybox one. So i figured it is a better fit here.

It must be explicitly called via mount.nfs4
```
mount.nfs4 192.168.0.108:/mnt/tank/movies /storage/videos/
```
and shows up via ```mount```
```
192.168.0.108:/mnt/tank/movies on /storage/videos type nfs4 (rw,relatime,vers=4.0,rsize=65536,wsize=65536,namlen=255,hard,proto=tcp,port=0,timeo=600,retrans=2,sec=sys,clientaddr=192.168.0.104,local_lock=none,addr=192.168.0.108)
```

The only thing I don't like is the patch ```nfs-utils-0001-fix-aclocal-libnfidmap.patch```.
It disables the aclocal macro to check for libnfsidmap but it's the only way I could get it to use the static lib.

This closes OpenELEC/OpenELEC.tv#2022

Note that this doesn't use any security (gss, kerberos, etc)
It would be nice if others could test this also.

### CODE REVIEW

**Comment by stefansaraev** [comment]
thanks. you should revert order of commits. nfs-utils should be last.

File: addons/networking/nfs-utils/package.mk [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  LDFLAGS="$LDFLAGS -static -lnfsidmap"
+}
+
+makeinstall_target() {
+  : # nop
+}
+
+addon() {
+  mkdir -p $ADDON_BUILD/$PKG_ADDON_ID/bin
+      cp -p $ROOT/$PKG_BUILD/utils/mount/mount.nfs $ADDON_BUILD/$PKG_ADDON_ID/bin
```
Comment (stefansaraev) [inline addons/networking/nfs-utils/package.mk]: install as mount.nfs4

cp -p $ROOT/$PKG_BUILD/utils/mount/mount.nfs $ADDON_BUILD/$PKG_ADDON_ID/bin/mount.nfs4

File: addons/networking/nfs-utils/package.mk [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+      ln -s mount.nfs $ADDON_BUILD/$PKG_ADDON_ID/bin/mount.nfs4
+      ln -s mount.nfs $ADDON_BUILD/$PKG_ADDON_ID/bin/umount.nfs
+      ln -s mount.nfs $ADDON_BUILD/$PKG_ADDON_ID/bin/umount.nfs4
```
Comment (stefansaraev) [inline addons/networking/nfs-utils/package.mk]: symlinks are not supported in addon zips. this increases addonzip size for (imo) no good reason. umount should not be necessary at all. busybox umount should handle umounting just fine. or it does not ?

**Comment by stefansaraev** [comment]
building tools/locktest fails on all supported projects except x86_64 (building on x86_64 here)

pick http://sprunge.us/BMYK into nfs-utils/patches

**Comment by lrusak** [comment]
Updated with the requested changes

**Comment by stefansaraev** [comment]
thanks. will do build test and merge later today. you forgot to move the first commit last when rebasing ? :D

**Comment by jpf91** [comment]
@lrusak As far as I can see Systemd mounts as described on http://wiki.openelec.tv/index.php?title=Mounting_network_shares always use the standard mount program. Because of this, it's not possible to use Systemd to mount nfs4 shares. Any idea how to fix this? Would it be possible to ship the util-linux mount program as an addon?

**Comment by lrusak** [comment]
Not possible, we use the busybox mount utility (not util-linux) and this doesn't have support for nfs4.

Not sure of anyway around this at the moment.

**Pull request merged**